src/components/App.jsx

```diff
@@ -89,42 +88,58 @@ const theme = createMuiTheme({
   }
 });
 
-const PrivateRoutes = () => (
-  <Switch>
-    <Redirect exact from="/" to="/events" />
-    <Route exact path="/events" component={EventsSection} />
-    <Route exact path="/settings" component={SettingsSection} />
-    <Route path="/events/:eventId/performances" component={PerformancesSection} />
-    <Route path="/events/:eventId/performance/:performanceId/adjudications" component={AdjudicationsSection} />
-    <Route component={Landing} />
-  </Switch>
-);
-
-const PublicRoutes = () => (
-  <Switch>
-    <Route exact path="/" component={Landing} />
-    <Route exact path={routes.SIGN_IN} component={SignInPage} />
-    <Route component={Landing} />
-  </Switch>
+const PrivateRoute = ({
+  component: Component,
+  path,
+  user,
+  ...rest
+}) => (
+  <Route
+    path={path}
+    render={props => (user ? <Component {...props} />
+      : <Redirect to={{ pathname: '/', state: { from: props.location }}} />)}
+    {...rest} />
 );
 
 export default class App extends React.Component {
-  state = {};
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      user: auth.currentUser,
+      loading: true
+    };
+  }
+
+  componentDidMount() {
+    auth.onAuthStateChanged((user) => {
+      this.setState({ user, loading: false });
+    });
+  }
+
 
   render() {
-    // TODO: check user login from Redux
-    const authenticated = true;
+    const { user, loading } = this.state;
     return (
       <MuiThemeProvider theme={theme}>
         <Router>
-          {authenticated ? (
-            <div>
-              <Header />
-              <PrivateRoutes />
-            </div>
-          ) : (
-            <PublicRoutes />
-          )}
+          <div>
+            { !loading && user && <Header /> }
+            {
+              // TODO: Once we merge in PR#27, we'll swap this with the react loading icon
+              loading ? <div> loading </div>
```

Comment:
is it possible to do something like 

```
loading ? (<div> loading </div>) : (
<React.Fragment>
   {user && <Header />}
   <Switch>
       ....
    </Switch>
</React.Fragment>
```

Makes the `loading` logic more readable instead of having `!loading` and then having it again at the bottom